### PR TITLE
Create dm-compatibility-catalog.md

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.1 (TiDB 6.1 versions)
- [x] v6.0 (TiDB 6.0 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/8481
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/dm-compatibility-catalog.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+summary: This document describes the compatibility between DM of different versions and upstream/downstream databases.
+---
+
+# Compatibility Catalog of TiDB Data Migration
+
+DM supports migrating data from different sources to TiDB clusters. Based on the data source type, DM has four compatibility levels:
+
+- **Generally available (GA)**: The application scenario has been verified and passed the GA test.
+- **Experimental**: Although the application scenario has been verified, the test does not cover all scenarios or involves only a limited number of users. The application scenario might encounter problems occasionally.
+- **Not tested**: DM is expected to be always compatible with MySQL during iteration. However, due to resource constraints, we cannot test all MySQL-derived versions. Therefore, the application is technically compatible with MySQL, but is not fully tested, which means you need to perform a test to ensure compatibility.
```
Comment (ran-huang) [inline dm/dm-compatibility-catalog.md:12]: @sunzhaoyang  What does "MySQL 衍生版本" mean exactly? I assume you mean "MySQL forks" such as MariaDB?
```suggestion
- **Not tested**: DM is expected to be always compatible with MySQL during iteration. However, due to resource constraints, not all MySQL forks are tested with DM. Therefore, the *not tested* source or target is technically compatible with DM, but is not fully tested, which means you need to verify its compatibility before you use.
```

File: dm/dm-compatibility-catalog.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+|MySQL 8.0|Experimental||
+|MariaDB < 10.1.2|Incompatible|Incompatible with binlog of the time type|
+|MariaDB 10.1.2 ~ 10.5.10|Experimental||
+|MariaDB > 10.5.10|Incompatible|Permission errors reported in the check procedure|
+
+## Target databases
+
+> **Warning:**
+>
+> DM 5.3.0 is not recommended because data replication fails with low probability if you have enabled GTID replication but do not enable relay log.
```
Comment (ran-huang) [inline dm/dm-compatibility-catalog.md:31]: It's recommended to put if clauses upfront. [ref](https://developers.google.com/style/clause-order)
```suggestion
> DM v5.3.0 is not recommended. If you have enabled GTID replication but do not enable relay log in DM v5.3.0, data replication fails with low probability.
```

File: dm/dm-compatibility-catalog.md (hunk 1 above)
Comment (sunzhaoyang) [reply dm/dm-compatibility-catalog.md:12]: Yes

**Review by sunzhaoyang — APPROVED** [review]
LGTM

**Review by ran-huang — APPROVED** [review]

**Pull request merged**